go/src/koding/kites/kloud/provider/provider.go

```diff
@@ -21,6 +21,10 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
+// All is a global lookup map used ny kloud to register
+// available stack providers.
+var All = make(map[string]func(*BaseProvider) stack.Provider)
```

Comment:
instead of exposing the var, i would expose a function 
smth like

```
type Stacker interface {
  Provide(*provider.BaseProvider) stack.Provider
}

func MustRegister(name string, s Stacker) {
  if _, ok := provider.All[name]; ok {
    panic("...")
  }

  all[name] = s  
}
```

it is not a problem, just a taste of my being on the safe side :)